generators/generator-base.js

```diff
@@ -1286,6 +1352,34 @@ module.exports = class extends PrivateBase {
      * @param {string} name - maven ArtifactId
      * @param {string} version - (optional) explicit dependency version number
      */
+    addGradleDependencyManagement(scope, group, name, version) {
+        const fullPath = 'build.gradle';
+        let dependency = `${group}:${name}`;
+        if (version) {
+            dependency += `:${version}`;
+        }
+        try {
+            jhipsterUtils.rewriteFile({
+                file: fullPath,
+                needle: 'jhipster-needle-gradle-dependency-management',
```

Comment:
where is the needle in gradle file for this?

Reply:
erf resolve conflict error.
I correct that tonight..

Reply:
what is the best way to add unit test on generated code ? sorry node noob :(

Reply:
have a look at the `test` directory, in `generator-jhipster`.
You can also generate tests in the application, so they match the generated code.